db/change_cache.go

```diff
@@ -215,6 +215,10 @@ func (c *changeCache) Start(initialSequence uint64) error {
 // Stops the cache. Clears its state and tells the housekeeping task to stop.
 func (c *changeCache) Stop() {
 
+	if c.IsStopped() {
+		return
+	}
```

Comment:
As discussed, there's still a race here if multiple callers invoke stop at the same time.  To avoid, replace c.IsStopped() with something that will set c.stopped, and return false/error if c.stopped is already true (and exit on that case).